server/src/main/java/org/gluu/oxtrust/api/RegistrationManagementService.java

```diff
@@ -0,0 +1,211 @@
+import org.slf4j.Logger;
+import org.xdi.config.oxtrust.AppConfiguration;
+import org.xdi.model.GluuAttribute;
+import org.xdi.util.StringHelper;
+
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.SecurityContext;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Implementation for register REST web services.
+ *
+ * @author Shoeb Khan
+ * @version July 06, 2018
+ */
+@Path(OxTrustApiConstants.BASE_API_URL + "/configurations/registration")
+@ProtectedApi
+public class RegistrationManagementService {
+
+    public static final String SERVER_DENIED_THE_REQUEST = "Server denied the request.";
+    public static final String MALFORMED_REQUEST = "The request is malformed.";
+
+    @Inject
+    private Logger log;
+
+    @Inject
+    private OrganizationService organizationService;
+
+    @Inject
+    private AttributeService attributeService;
+
+    @Inject
+    private JsonConfigurationService jsonConfigurationService;
+
+
+    /**
+     * This operation retrieves the configuration info for the specified findAttributesForPattern criteria,
+     * or all the configurations if no findAttributesForPattern criteria specified.
+     * <p>
+     * Note:
+     * Due to some properties of {@link GluuAttribute} class, direct JSON mapping may result in JSON exceptions.
+     * This may not be an issue on browser side, however, the java clients that are calling this method
+     * should not rely on automatic conversion of JSON objects to {@link RegistrationManagementResponse} class
+     * and should rather use {@link org.codehaus.jackson.map.DeserializationConfig.Feature} class <code>FAIL_ON_UNKNOWN_PROPERTIES</code> properties
+     * for deserialization.
+     *
+     * @param searchPattern   Search pattern.
+     * @param securityContext An injectable interface that provides access to security related information.
+     * @return response
+     */
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @ApiOperation(
+            value = "Reads registration configuration info.",
+            notes = "Reads registration configuration info."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(code = 403, message = "access_denied\n" +
+                    SERVER_DENIED_THE_REQUEST),
+            @ApiResponse(code = 200, response = RegistrationManagementResponse.class, message = "Success")
+    })
+    public Response getConfiguration(@QueryParam(value = "searchPattern") String searchPattern, @HeaderParam("Authorization") String authorization, @Context HttpServletRequest httpRequest, @Context SecurityContext securityContext) {
```

Comment:
Of all these arguments, you're only using the "searchPattern", are they required or can we remove them ?